src/content/reference/react/captureOwnerStack.md

```diff
@@ -0,0 +1,301 @@
+
+`SubComponent` is omitted because it's already part of the callstack.
+
+</DeepDive>
+
+## Usage {/*usage*/}
+
+### Improve error reporting {/*improve-error-reporting*/}
+
+Check out the following example to see how to use `captureOwnerStack` to improve error reporting:
+
+- [createRoot usage: Show a dialog for uncaught errors
+](/reference/react-dom/client/createRoot#show-a-dialog-for-uncaught-errors)
+- [createRoot usage: Displaying Error Boundary errors](/reference/react-dom/client/createRoot#show-a-dialog-for-uncaught-errors)
+- [createRoot usage: Displaying a dialog for recoverable errors](/reference/react-dom/client/createRoot#displaying-a-dialog-for-recoverable-errors)
+
+### Expanding error stacks {/*expanding-error-stacks*/}
+
+In addition to the <CodeStep step={1}>stack trace of the error</CodeStep> itself, you can use <CodeStep step={2}>`captureOwnerStack`</CodeStep> to append the Owner Stack.
+This can help trace the error especially when the error is caused by props. The Owner Stack helps trace the flow of props.
+
+
+```js src/index.js [[1, 8, "error.stack"], [2, 7, "captureOwnerStack()"]]
+import {captureOwnerStack} from 'react';
+import {createRoot} from 'react-dom/client';
+
+const root = createRoot(document.getElementById('root'), {
+  onUncaughtError: (error, errorInfo) => {
+    if (process.env.NODE_ENV !== 'production') {
+      const ownerStack = captureOwnerStack();
+      error.stack += ownerStack;
+    }
+
+    console.error('Uncaught', error);
+  },
+}).render(<App />);
+```
+
+<Sandpack>
+
+```js
+function useCustomHook() {
+  throw new Error('Boom!');
+}
+
+function Component() {
+  useCustomHook();
+}
+
+export default function App() {
+  return <Component />;
+}
+```
+
+```js src/index.js
+import {captureOwnerStack} from 'react';
+import {createRoot} from 'react-dom/client';
+import App from './App.js';
+import './styles.css';
+
+const root = createRoot(document.getElementById('root'), {
+  onUncaughtError: (error, errorInfo) => {
+    if (process.env.NODE_ENV !== 'production') {
+      const ownerStack = captureOwnerStack();
+      error.stack =
+        // The stack is only split because these sandboxes don't implement ignore-listing 3rd party frames via sourcemaps.
+        // A framework would ignore-list stackframes from React via sourcemaps and then you could just `error.stack += ownerStack`.
+        // To learn more about ignore-listing see https://developer.chrome.com/docs/devtools/x-google-ignore-list
+        error.stack.split('\n    at react-stack-bottom-frame')[0] + ownerStack;
+    }
+
+    // The stacks are logged instead of showing them in the UI directly to highlight that browsers will apply sourcemaps to the logged stacks.
+    // Note that sourcemapping is only applied in the real browser console not in the fake one displayed on this page.
+    console.error('Uncaught', error);
```

Comment:
If you've patched `console.error` to add owner stacks, and you call console.error here, it will result in multiple owner stacks being applied. It's not clear to me whether you should only patch console.error, or if you should instrument the root handlers and console.error without them overlapping somehow. 

@sebmarkbage how should this work ideally? Maybe we're over-emphasizing the root handler case?

The way I expected this to work is like:

```
onUncaughtError() {
  if (process.env.NODE_ENV !== 'production') {
      const ownerStack = captureOwnerStack();
      sendDevOnlyLogs(error, ownerStack);
  } else {
    sendProductionLogs(error);
  }
  
  // Patch or DevTools will add logs to the console output.
  console.error(error);
}
```

Reply:
Yeah I think the usage in error handlers is the better example. If there's a need to document polyfilling what React Devtools is doing, we can add it later.

Reply:
@eps1lon how do you implement a dev dialog without patching console.error? You can't only do the root handlers because you'll miss react warnings like key errors.

Reply:
This seems like a separate task and not strictly tied to `captureOwnerStack`. We haven't documented how to implement a dev dialog that displays `console.error` before so I don't see why this needs to be coupled with `captureOwnerStack`. 

In Next.js we're just reading errors from console. Owner stacks are then always displayed in the overlay via `captureOwnerStack` (no option). But displaying Component stacks directly in the console is left to React DevTools.

Reply:
> We haven't documented how to implement a dev dialog that displays console.error

Yeah but we have for the other root handlers, and if you did that plus your own console.error patch before, then owner stacks break that implementation so you need docs for how to add it back. So we need to at least tell people how to get back to what worked before owner stacks.